integration_tests/test_solo_machine.py

```diff
@@ -0,0 +1,182 @@
+        self.grpc_port = ports.grpc_port(base_port)
+        self.rpc_port = ports.rpc_port(base_port)
+        self.bin_dir = "/tmp/solomachine"
+        self.bin_file = os.path.join(self.bin_dir, "solo-machine")
+        self.sign_file = None
+
+        self.trusted_height = None
+        self.trusted_hash = None
+
+    def get_chain_info(self):
+        result = requests.get(f"http://127.0.0.1:{self.rpc_port}/block").json()
+        self.trusted_hash = result["result"]["block_id"]["hash"]
+        self.trusted_height = result["result"]["block"]["header"]["height"]
+
+    def set_env(self):
+        os.environ["SOLO_DB_URI"] = "sqlite:///tmp/solo-machine.db"
+        os.environ["SOLO_SIGNER"] = self.sign_file
+        os.environ[
+            "SOLO_MNEMONIC"
+        ] = "awesome there minute cash write immune tag \
+reopen price congress trouble reunion south wisdom donate credit below leave \
+wisdom eagle sail siege rice train"
+        os.environ["SOLO_HD_PATH"] = "m/44'/394'/0'/0/0"
+        os.environ["SOLO_ACCOUNT_PREFIX"] = "cro"
+        os.environ["SOLO_ADDRESS_ALGO"] = "secp256k1"
+        os.environ["SOLO_FEE_AMOUNT"] = "1000"
+        os.environ["SOLO_FEE_DENOM"] = "basecro"
+        os.environ["SOLO_GAS_LIMIT"] = "300000"
+        os.environ["SOLO_GRPC_ADDRESS"] = f"http://0.0.0.0:{self.grpc_port}"
+        os.environ["SOLO_RPC_ADDRESS"] = f"http://0.0.0.0:{self.rpc_port}"
+        os.environ["SOLO_TRUSTED_HASH"] = self.trusted_hash
+        os.environ["SOLO_TRUSTED_HEIGHT"] = self.trusted_height
+
+    def download_bin_file(self):
+        sysstr = platform.system()
+        if sysstr == "Darwin":
+            url = "https://github.com/crypto-com/ibc-solo-machine/releases/\
+download/v0.1.1/macos-latest-v0.1.1.tar.gz"
+            self.sign_file = os.path.join(self.bin_dir, "libmnemonic_signer.dylib")
+        else:
+            url = "https://github.com/crypto-com/ibc-solo-machine/releases/\
+download/v0.1.1/ubuntu-latest-v0.1.1.tar.gz"
+            self.sign_file = os.path.join(self.bin_dir, "libmnemonic_signer.so")
+        local_filename = url.split("/")[-1]
+        tar_file = f"/tmp/{local_filename}"
+        if os.path.exists(self.bin_file) and os.path.exists(self.sign_file):
+            return
+        with requests.get(url, allow_redirects=True, stream=True) as r:
+            r.raise_for_status()
+            with open(tar_file, "wb") as f:
+                for chunk in r.iter_content(chunk_size=1024 * 8):
+                    f.write(chunk)
+        # uncompress
+        file = tarfile.open(tar_file)
+        file.extractall(self.bin_dir)
+        file.close()
+        # check file exists
+        if not os.path.exists(self.bin_file):
+            raise Exception(f"solomachine binary file {self.bin_file} is not exists")
+        if not os.path.exists(self.sign_file):
+            raise Exception(
+                f"solomachine signature file {self.sign_file} is not exists"
+            )
```

Comment:
probably ok, but it seems a bit cumbersome -- it can be replaced by having it fetched in the Nix environment (`builtins.fetchTarball`) and having the solo machine binary in the environment's PATH

Reply:
sure